### PR TITLE
Fix: Permission Handling Issues in Microphone and Camera Capture Dialogs

### LINKED ISSUE #8829: Permission Handling Issues in Microphone and Camera Capture Dialogs
## **Describe the Bug**
These issues are for `Sample Test Details` page:
1. In `Record audio` modal,
    - [ ] `Recording started timer` starts before giving permission of microphone access in browser.
Image

2. In `Open Camera` modal,    
    - [ ] After blocking access of camera , Empty container is there.
 Image

    - [ ] After clicking on `Switch` button, Camera stuck. 

Camera and microphone permissions are not handled correctly. When the user does not grant permission for microphone or camera, the app does not handle the error appropriately, and permission errors are not clearly communicated to the user.

## **To Reproduce**
1. Click on `Sample Test` in sidebar.
2. Click on any item's `Sample Details`.
3. Go to the Audio Capture dialog or Camera Capture dialog.
4. Deny the microphone or camera permission prompt.
5. Observe that the recording timer starts before microphone permission is granted, and after blocking camera access, an empty container appears with the camera freezing upon clicking the switch button.

## **Expected Behavior**
- In the `Record audio` modal, the recording timer should start only after the user grants microphone access. If permission is denied, an appropriate error message should be displayed to the user.
- In the `Open Camera` modal, if camera access is blocked, a proper error message should be shown instead of an empty container. The camera should not freeze after clicking the `Switch` button, and a clear message should be displayed if the camera cannot be switched.

### Record Audio modal:
https://github.com/user-attachments/assets/d3fe377e-5a98-4f9e-9c79-5225f29553fd
## **Desktop (please complete the following information):**
- OS: Windows
- Browser: Brave
- Version: 1.71.114

### PR DESCRIPTION
**Pull request by shauryag2002**
## Proposed Changes

- Fixes #8829 
- In the `Record audio` modal, the recording timer now starts only after the user grants microphone access. If permission is denied, an appropriate error message is displayed to the user.
- In the `Open Camera` modal, when camera access is blocked, a proper error message is shown instead of an empty container. The camera no longer freezes after clicking the `Switch` button, and a clear message is displayed if the camera cannot be switched.

## **Video**:

https://github.com/user-attachments/assets/39556953-d678-4e18-8296-c6155a39bf86

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by nihal467** [comment]
LGTM

File: src/Components/Files/AudioCaptureDialog.tsx [hunk 1]
```diff
@@ -35,9 +36,19 @@ export default function AudioCaptureDialog(props: AudioCaptureDialogProps) {
...
+      .getUserMedia({ audio: true })
+      .then(() => {
+        setStatus("RECORDING");
+        startRecording();
+        timer.start();
+      })
+      .catch(() => {
+        Notify.Error({
+          msg: "Please grant microphone permission to record audio.",
+        });
```
Comment (khavinshankar) [inline src/Components/Files/AudioCaptureDialog.tsx]: i18n

**Review by khavinshankar — CHANGES_REQUESTED** [review]
LGTM, Nicely done, just a minor i18n change

File: src/Components/Files/AudioCaptureDialog.tsx (hunk 1 above)
Comment (shauryag2002) [reply src/Components/Files/AudioCaptureDialog.tsx]: Done

**Review by bodhish — APPROVED** [review]
LGTM

**Pull request merged**